### ISSUE TITLE
Bug in jsx renderer rendering style attribute

### ISSUE THREAD

**Issue by Mushus** [honojs/hono#2388]
### What version of Hono are you using?

4.1.2

### What runtime/platform is your app running on?

Node v20.11.0

### What steps can reproduce the bug?

Execute the following script:
```ts
import { Hono } from "hono";
import React from "hono/jsx"

const app = new Hono().get("/", (c) => c.html(<div style={{"font": '"NotoSans"'}}/>));

// expect: <html><div style="font:&quot;NotoSans&quot;"></div></html>
// actual: <html><div style="font:"NotoSans""></div></html>
console.log(await (await app.request("/")).text());
```

### What is the expected behavior?

print `<html><div style="font:&quot;NotoSans&quot;"></div></html>`

### What do you see instead?

print `<html><div style="font:"NotoSans""></div></html>`

**Comment by NicoPlyley** [comment]
Is there a reason you expect it this way? Wouldn't the proper syntax be `<div style={{"fontFamily": "'NotoSans'"}} />` to use single quotes inside

**Comment by Mushus** [comment]
There appears to be a difference in the behavior of whether the style attribute is escaped when passing a string type and when passing a Record<string, string> type.
Failure to escape attribute values may cause cross-site scripting vulnerabilities.

```ts
import { Hono } from "hono";
import React from "hono/jsx"

{ // escaped
  const app = new Hono().get("/", (c) => c.html(<div style={'"'}/>));
  // actual: <div style="&quot;"></div>
  console.log(await (await app.request("/")).text());
}

{ // not escaped
  const app = new Hono().get("/", (c) => c.html(<div style={{ "": '"' }}/>));
  // actual: <div style="&quot;"></div>
  console.log(await (await app.request("/")).text());
}

{ // xss
  const app = new Hono().get("/", (c) => c.html(<div style={{ "": '" onclick="alert(&quot;not escaped&quot;)' }}/>));
  // actual: <div style="&quot;"></div>
  console.log(await (await app.request("/")).text());
}

{ // xss
  const app = new Hono().get("/", (c) => c.html(<div style={{ "": '"> <script>alert("not escaped")</script><a a="'}}/>));
  // actual: <div style="&quot;"></div>
  console.log(await (await app.request("/")).text());
}
```

**Comment by usualoma** [comment]
Hi @Mushus 
Thank you! I think your point is right on. This is a bug and will be fixed in #2390.
